db/db.go

```diff
@@ -97,8 +98,12 @@ func withinTxImpl(db DB, beginStrategy func(db DB) (transaction.Transaction, err
 			return err
 		}
 
-		log.Warning("scoped transaction deadlocked, retrying %d / %d", attempt, db.Options().RetryTxCount)
-		time.Sleep(db.Options().RetryTxInterval)
+		retryInterval := db.Options().RetryTxInterval
```

Comment:
this function violates SRP. I suggest introducing `func getRetryInterrval(options.Options) time.Duration` or `func getRetryInterrval(DB) time.Duration`